qiskit/extensions/standard/t.py

```diff
@@ -78,20 +89,35 @@ def inverse(self):
         return TGate()
 
     def to_matrix(self):
-        """Return a Numpy.array for the S gate."""
+        """Return a numpy.array for the inverse T gate."""
         return numpy.array([[1, 0],
-                            [0, (1-1j) / numpy.sqrt(2)]], dtype=complex)
+                            [0, (1 - 1j) / numpy.sqrt(2)]], dtype=complex)
+
+
+class TdgGate(TInvGate, metaclass=TInvMeta):
+    """The deprecated TInv gate."""
+
+    def __init__(self):
+        warnings.warn('TdgGate is deprecated, use TInvGate instead!', DeprecationWarning, 2)
+        super().__init__()
 
 
 def t(self, q):  # pylint: disable=invalid-name
     """Apply T to q."""
     return self.append(TGate(), [q], [])
 
 
+def tinv(self, q):
+    """Apply TInv to q."""
+    return self.append(TInvGate(), [q], [])
+
+
 def tdg(self, q):
-    """Apply Tdg to q."""
+    """Apply Tdg (deprecated!) to q."""
+    warnings.warn('tdg() is deprecated, use tinv() instead!', DeprecationWarning, 2)
```

Comment:
Are `tdg` and `sdg` going to be kept as aliases?

Reply:
We can also keep them, I don't have such a strong opinion on this. 👍